legion/engine/physics/physx/physx_event_process_funcs.cpp

```diff
@@ -86,18 +117,17 @@ namespace legion::physics
     {
```

Comment:
there are many similar instructions (e.g. static casting to PxRigidDynamic) within these functions but they achieve different things `processAngularDragModification` modifies angular drag, `processVelocityModification` modifies velocity, etc. I would make the argument that these functions have been refactored to a reasonable extent. (e.g. `calculateLocalColliderTransform` ). As for the addFirst and addNext, they exist because these lead to different code paths (If its the first collider to be placed an entire PxRigidDynamic needs to instantiated but if its the next one we only need to instantiate a PxShape).